src/is-address-ipv6.cpp

```diff
@@ -0,0 +1,68 @@
+// @file is-address-ipv6.c
+//
+// This tool takes an IP address as a string, and returns 0 if it is IPv6
```

Comment:
Technically, it takes an input string and returns 0 if that's an IPv6 address. (The distinction is that it's not user error to pass a hostname to this function.)

The Python script also had a helpful note about square brackets not actually being part of an IPv6 address and a warning that this script would accordingly return 1.